src/Components/EmailNotificationForm/EmailNotificationForm.js

```diff
@@ -37,28 +43,48 @@ class EmailNotificationForm extends Component {
         email: this.state.email,
         language: this.state.language,
       }
-      // this.props.postNewEmailSubscriber(newEmailSubscriber);
+
+      postNewEmailSubscriber(newEmailSubscriber)
+        .then(response => {
+          this.setState({ isSubmitting: true });
+          if (response.status !== 200) {
+            throw new Error (`${response.status} : Sorry, cannot fetch the data.`)
+          }
+          if (!response.ok) {
+            throw new Error ('Something has gone wrong, please try again.')
+          }
```

Comment:
What do you think about just using lines 53-55 (i.e. just using `!response.ok`) to throw an Error? I believe it would catch what lines 50-52 are doing as well.

Reply:
We need to double check this with the backend.  I don't know if they are setting response.ok to false at any point.  I would prefer to just check for one, but it depends on the response object!

Reply:
Good point! We can definitely verify this with our BE teammates.